Subject: generator/customize.ml: Split --chown parameter on ':' character
From: Richard W.M. Jones rjones@redhat.com Fri Jan 19 13:18:00 2024 +0000
Date: Fri Jan 19 13:24:37 2024 +0000:
Git: e9a728bb22dfa1de5328fbbe7d6d7acad2dbed64

The previous code split it on ',' which was completely wrong.
(It reveals the lack of testing however).

Fixes: commit c08032ebe2763f5e9ce5b14e003721475219d390
Reported-by: Yongkui Guo

diff --git a/generator/customize.ml b/generator/customize.ml
index e64b45c0d..e2c4b6051 100644
--- a/generator/customize.ml
+++ b/generator/customize.ml
@@ -775,7 +775,7 @@ let rec argspec () =
     let len = String.length arg in
     String.sub arg 0 i, String.sub arg (i+1) (len-(i+1))
   and split_string_triplet option_name arg =
-    match String.nsplit ~max:3 \",\" arg with
+    match String.nsplit ~max:3 \":\" arg with
     | [a; b; c] -> a, b, c
     | _ ->
         error (f_\"invalid format for '--%%s' parameter, see the man page\")
